### PR TITLE
Upsert flags when loading feature flags with computed errors

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context
https://github.com/PostHog/posthog-js-lite/blob/4a707fdc660785c187e0bfd99ec2df1887afcf5c/posthog-core/src/index.ts#L931
I did the wrong check.
Refactor for reading the JSON from resources instead of strings, its easier to maintain and read due to identation.

## :green_heart: How did you test it?
Unit tests

## :pencil: Checklist

- [X] I reviewed the submitted code.
- [X] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [X] No breaking change or entry added to the changelog.

### CODE REVIEW

File: posthog/src/test/java/com/posthog/internal/PostHogFeatureFlagsTest.kt [hunk 1]
```diff
@@ -131,7 +134,7 @@ internal class PostHogFeatureFlagsTest {
     }
 
     @Test
-    fun `merge feature flags if no errors`() {
+    fun `merge feature flags if there are errors`() {
```
Comment (pauldambra) [inline posthog/src/test/java/com/posthog/internal/PostHogFeatureFlagsTest.kt:137]: 🤯 😊

File: posthog/src/test/resources/json/basic-decide-no-errors.json
Comment (pauldambra) [inline posthog/src/test/resources/json/basic-decide-no-errors.json:1]: I'd find these two easier to think about if they were generated by a method. So I could see the difference is the payload and the `errorsWhileComputingFlags`

But it's stylistic and easy to change so non-blocking

**Review by pauldambra — APPROVED** [review]
👍

File: posthog/src/test/resources/json/basic-decide-no-errors.json
Comment (marandaneto) [reply posthog/src/test/resources/json/basic-decide-no-errors.json:1]: yeah, I agree, that's why I named the files this way, it's a bit more obvious but there's no syntax highlighting for JSONs represented as Strings so it's also not ideal.

**Pull request merged**